File: mobile/lib/features/collections/models/pickup_request.dart
class PickupRequest {
  const PickupRequest({
    required this.id,
    required this.recoveryProposalId,
    required this.collectionSlotId,
    this.collectorId,
    required this.ownerId,
    required this.scheduledStart,
    required this.scheduledEnd,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String recoveryProposalId;
  final String collectionSlotId;
  final String? collectorId;
  final String ownerId;
  final DateTime scheduledStart;
  final DateTime scheduledEnd;
  final String status;
  final DateTime createdAt;
  final DateTime updatedAt;

  factory PickupRequest.fromJson(Map<String, dynamic> json) {
    return PickupRequest(
      id: json['id'] as String,
      recoveryProposalId: json['recoveryProposalId'] as String,
      collectionSlotId: json['collectionSlotId'] as String,
      collectorId: json['collectorId'] as String?,
      ownerId: json['ownerId'] as String,
      scheduledStart: DateTime.parse(json['scheduledStart'] as String),
      scheduledEnd: DateTime.parse(json['scheduledEnd'] as String),
      status: json['status'] as String,
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: DateTime.parse(json['updatedAt'] as String),
    );
  }

  Map<String, dynamic> toJson() => <String, dynamic>{
    'id': id,
    'recoveryProposalId': recoveryProposalId,
    'collectionSlotId': collectionSlotId,
    'collectorId': collectorId,
    'ownerId': ownerId,
    'scheduledStart': scheduledStart.toIso8601String(),
    'scheduledEnd': scheduledEnd.toIso8601String(),
    'status': status,
    'createdAt': createdAt.toIso8601String(),
    'updatedAt': updatedAt.toIso8601String(),
  };
}

File: backend/src/WasteToValue.Api/Modules/Collections/Entities/PickupRequest.cs
namespace WasteToValue.Api.Modules.Collections.Entities;

public sealed class PickupRequest
{
    public Guid Id { get; set; }
    public Guid RecoveryProposalId { get; set; }
    public Guid CollectionSlotId { get; set; }
    public Guid? CollectorId { get; set; }
    public Guid OwnerId { get; set; }
    public string PickupAddressEncrypted { get; set; } = string.Empty;
    public DateTimeOffset ScheduledStart { get; set; }
    public DateTimeOffset ScheduledEnd { get; set; }
    public string Status { get; set; } = "CONFIRMED";
    public string? VerificationCode { get; set; }
    public string IdempotencyKey { get; set; } = string.Empty;
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }
    public int Version { get; set; } = 1;

    public CollectionSlot? CollectionSlot { get; set; }
    public ICollection<PickupEvent> Events { get; set; } = new List<PickupEvent>();
    public ICollection<HandoverProof> HandoverProofs { get; set; } = new List<HandoverProof>();
}
